tests/scipy_optimize_test.py

```diff
@@ -86,6 +86,14 @@ def min_op(x0):
     scipy_res = scipy.optimize.minimize(func(np), x0, method='BFGS').x
     self.assertAllClose(scipy_res, jax_res, atol=2e-5, check_dtypes=False)
 
+  def test_fixes4594(self):
+    n = 2
+    A = jnp.eye(n) * 1e4
+    def f(x):
+      return jnp.mean((A @ x) ** 2)
+    results = jax.scipy.optimize.minimize(f, jnp.ones(n), method='BFGS')
+    assert jnp.isclose(results.x, jnp.zeros(n)).all()
```

Comment:
This test fails on TPU; we'll need to change the precision

Also, since this is in the test framework, we should be using `self.assertAllClose`. If it had been using that, I'd have been able to tell you what precision we need 😀

Reply:
Oops, I changed it now to try with 1e-8.